add functionality for deactivate tariff for tree cases

LiliaMokhnatska opened the pull request:
# Green City UBS  Deactivaite tariff new cases 


## Summary Of Issue :
400 response in when diactivate tariff with parameters: 1) regionsId, stationsId; 2) citiesId, courierId, regionsId; 3) courierId, regionsId, stationsId 


It is expected that the 200 response is apear  

## Issue Link :
https://github.com/ita-social-projects/GreenCity/issues/5187

## Summary Of Changes :

1. add methods deactivateTariffsByRegionAndReceivingStations, deactivateTariffsByCourierAndRegionAndCities and deactivateTariffsByCourierAndRegionAndReceivingStations into DeactivateChosenEntityRepository.java
2. add methods shouldDeactivateTariffsByRegionAndReceivingStations, shouldDeactivateTariffsByCourierAndRegionAndCities, shouldDeactivateTariffsByCourierAndRegionAndReceivingStations into SuperAdminServiceImpl.java
3. change deactivateTariffForChosenParam method in SuperAdminServiceImpl.java
4. add tests for these cases in SuperAdminServiceImplTest.java



## How to test :
1. Admin need to be logged in http://localhost:8060/swagger-ui.html#/own-security-controller
2. Register Token 
3. Go to Main UBS menu -> Choose "Тарифи”
4. Click on tools “Gear”-> item “Деактивувати”.
5. Choose next parameters:
1) regionsId, stationsId;
or
2) citiesId, courierId, regionsId;
or
3) courierId, regionsId, stationsId





## CHECK LIST
- [x] Code is up-to-date with the `dev` branch.
- [x] You've successfully built and run the tests locally.
- [x] There are new or updated unit tests validating the change.
- [x] JIRA/ Github Issue number & title in PR title (ISSUE-#5058: Ticket title)
- [x] This template filled (above this section).
- [ ] Sonar's report does not contain bugs, vulnerabilities, security issues, code smells ar duplication
- [x] `NEED_REVIEW` and `READY_FOR_REVIEW` labels are added.
- [x] All files reviewed before sending to reviewers

LiliaMokhnatska committed:
add functionality for deactivate tariff foe tree cases

RomaMocherniuk commented on dao/src/main/java/greencity/repository/DeactivateChosenEntityRepository.java:
```diff
@@ -188,6 +188,71 @@ public void deactivateTariffsByRegionAndCitiesAndStations(Long regionId, List<Lo
 
+    /**
+     * Method that deactivate tariffs for region id, list of cities id , list of
+     * station id and courier id.
```
Javadoc does not describe what the method does

LiliaMokhnatska replied:
changed

RomaMocherniuk commented on dao/src/main/java/greencity/repository/DeactivateChosenEntityRepository.java:
```diff
@@ -188,6 +188,71 @@ public void deactivateTariffsByRegionAndCitiesAndStations(Long regionId, List<Lo
+            .setParameter(REGION_ID, regionId)
+            .setParameter(STATIONS_ID, stationsId)
+            .getResultList()
+            .forEach(tariffsInfo -> tariffsInfo.setLocationStatus(LocationStatus.DEACTIVATED));
```
Can we use `UPDATE` here?

LiliaMokhnatska replied:
Ok. I'll try to use update.

RomaMocherniuk commented on dao/src/main/java/greencity/repository/DeactivateChosenEntityRepository.java:
```diff
@@ -188,6 +188,71 @@ public void deactivateTariffsByRegionAndCitiesAndStations(Long regionId, List<Lo
+
+    /**
+     * Method that deactivate tariffs for region id, list of cities id and courier
+     * id.
```
please mode `id` to the previous line

RomaMocherniuk commented on dao/src/main/java/greencity/repository/DeactivateChosenEntityRepository.java:
```diff
@@ -188,6 +188,71 @@ public void deactivateTariffsByRegionAndCitiesAndStations(Long regionId, List<Lo
+            .setParameter(CITIES_ID, citiesId)
+            .setParameter(COURIER_ID, courierId)
+            .getResultList()
+            .forEach(tariffsInfo -> tariffsInfo.setLocationStatus(LocationStatus.DEACTIVATED));
```
same here

RomaMocherniuk commented on dao/src/main/java/greencity/repository/DeactivateChosenEntityRepository.java:
```diff
@@ -188,6 +188,71 @@ public void deactivateTariffsByRegionAndCitiesAndStations(Long regionId, List<Lo
+
+    /**
+     * Method that deactivate tariffs for region id, list of station id and courier
+     * id.
```
same here

RomaMocherniuk commented on dao/src/main/java/greencity/repository/DeactivateChosenEntityRepository.java:
```diff
@@ -188,6 +188,71 @@ public void deactivateTariffsByRegionAndCitiesAndStations(Long regionId, List<Lo
+            .setParameter(STATIONS_ID, stationsId)
+            .setParameter(COURIER_ID, courierId)
+            .getResultList()
+            .forEach(tariffsInfo -> tariffsInfo.setLocationStatus(LocationStatus.DEACTIVATED));
```
same here

RomaMocherniuk commented on service/src/main/java/greencity/service/ubs/SuperAdminServiceImpl.java:
```diff
@@ -96,6 +96,13 @@ public class SuperAdminServiceImpl implements SuperAdminService {
     private static final String RECEIVING_STATIONS_OR_COURIER_EXIST_MESSAGE =
         "Current receiving stations: %s or courier: %s don't exist.";
     private static final String REGION_OR_COURIER_EXIST_MESSAGE = "Current region: %s or courier: %s don't exist.";
+    private static final String REGION_OR_RECEIVING_STATIONS_EXIST_MESSAGE =
```
I see in the constant name `STATIONS_EXIST` and in the message that `stations: %s don't exist`

LiliaMokhnatska replied:
Yes, I change it

RomaMocherniuk commented on service/src/main/java/greencity/service/ubs/SuperAdminServiceImpl.java:
```diff
@@ -96,6 +96,13 @@ public class SuperAdminServiceImpl implements SuperAdminService {
     private static final String REGION_OR_COURIER_EXIST_MESSAGE = "Current region: %s or courier: %s don't exist.";
+    private static final String REGION_OR_RECEIVING_STATIONS_EXIST_MESSAGE =
+        "Current region: %s or receiving stations: %s don't exist.";
+    private static final String REGION_OR_CITIES_OR_COURIER_EXIST_MESSAGE =
```
same here

RomaMocherniuk commented on service/src/main/java/greencity/service/ubs/SuperAdminServiceImpl.java:
```diff
@@ -96,6 +96,13 @@ public class SuperAdminServiceImpl implements SuperAdminService {
+    private static final String REGION_OR_CITIES_OR_COURIER_EXIST_MESSAGE =
+        "Current region: %s or cities: %s or courier: %s don't exist.";
+
+    private static final String REGION_OR_RECEIVING_STATIONS_OR_COURIER_EXIST_MESSAGE =
```
same here

RomaMocherniuk commented on service/src/main/java/greencity/service/ubs/SuperAdminServiceImpl.java:
```diff
@@ -858,6 +874,101 @@ private boolean shouldDeactivateTariffsByCourierAndReceivingStations(DetailsOfDe
+
+    /**
+     * Method that checks if the tariff should be deactivated by courier id, regions
+     * id and receiving stations id. In this case size of RegionsList should be one
```
same here

RomaMocherniuk commented on service/src/main/java/greencity/service/ubs/SuperAdminServiceImpl.java:
```diff
@@ -858,6 +874,101 @@ private boolean shouldDeactivateTariffsByCourierAndReceivingStations(DetailsOfDe
+
+    /**
+     * Method that checks if the tariff should be deactivated by courier id, regions
+     * id and cities. In this case size of RegionsList should be one because we
```
move `id` to the previous line

LiliaMokhnatska committed:
change methods deactivateTariffsByRegionAndReceivingStations
deactivateTariffsByCourierAndRegionAndCities
deactivateTariffsByCourierAndRegionAndReceivingStations
in repository
change names from stationsId, citiesId,  regionsId stationsIds, citiesIds,  regionsIds
where it was needed

LiliaMokhnatska committed:
1) change name of EXIST messages
2) add new conditions in method
deactivateTariffForChosenParam
in SuperAdminServiceImpl

3) add tests for new conditions

LiliaMokhnatska committed:
2) change method
deactivateTariffForChosenParam
in SuperAdminServiceImpl

LiliaMokhnatska committed:
2) change method
deactivateTariffForChosenParam
in SuperAdminServiceImpl

LiliaMokhnatska committed:
2) change get() on orElseThrow ()
in SuperAdminServiceImpl

LiliaMokhnatska committed:
change get() on orElseThrow ()
in SuperAdminServiceImpl

LiliaMokhnatska committed:
change get() on orElseThrow ()
in SuperAdminServiceImpl

LiliaMokhnatska committed:
one more change get() on orElseThrow ()
in SuperAdminServiceImpl

LiliaMokhnatska committed:
change orElseThrow() on  get()
in SuperAdminServiceImpl
